addons/web_editor/static/lib/odoo-editor/test/spec/color.test.js

```diff
@@ -39,4 +39,22 @@ describe('applyColor', () => {
             contentAfter: '<p>ab<font style="background-color: rgb(255, 0, 0);">[]\u200B</font>cd</p>',
         });
     });
+    it('should apply a color on empty selection', async () => {
+        await testEditor(BasicEditor, {
+            contentBefore: '<p>[<br></p><p><br></p><p>]<br></p>',
```

Comment:
Thanks for adding tests! It's a little weird to color a `<br/>` but I guess it makes sense for this case where we actually selected them and nothing else. I'm a little worried of potential consequences of wrapping a placeholder line break like this in a `<font>` because this is not a case we usually have and I don't know how well we handle it. So I'd make sure to test this manually quite deeply (doing multiple shift+enter and enter combinations on the colored paragraphs, navigating with the arrow keys, writing and deleting text, and removing the styles).

Apart from that, it seems to me that the original traceback should be fixed at its root, which is that the `_processAndApplyColor` method of `wysiwyg` expects this `applyColor` command to return an array with at least one element (which can't and shouldn't be guaranteed). So the line that says `} else if (!this.lastMediaClicked) {` in `web_editor/static/src/js/wysiwyg/wysiwyg.js` should be changed to `} else if (!this.lastMediaClicked && coloredElements.length) {`. That fix would remove the traceback without changing the behavior of the editor.

I'll let @dmo-odoo decide whether we should only do that latter fix or if he's ok with this commit (which might be an improvement but it's a behavior change so there's a choice involved :-)) as well.

Thanks for your work in any case :+1:

Reply:
I also think coloring the BR is weird but let's start by fixing the traceback.

Reply:
I double checked this PR and I agree with @Zinston. I think coloring the BR does not match the expectations of what is usually done in the editor. I have made a suggestion below regarding what the expected result might be.